Барроуз-Уилер/BWT.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Linq;
+
+public class BWT
+{
+    public static int CompareShifts(string input, int index1, int index2)
+    {
+        string shift1 = input.Substring(index1) + input.Substring(0, index1); 
+        string shift2 = input.Substring(index2) + input.Substring(0, index2);
+
+        return String.Compare(shift1, shift2);
+    }
+
+    public static (string, int) BWTransform(string input)
+    {
+        var suffixArray = Enumerable.Range(0, input.Length).ToArray();
+        Array.Sort(suffixArray, (index1, index2) => BWT.CompareShifts(input, index1, index2));
+
+        for (int i = 0; i < suffixArray.Length; ++i)
+        {
+            suffixArray[i] = i;
+        }
+
+        Array.Sort(suffixArray, (index1, index2) => BWT.CompareShifts(input, index1, index2));
+
+        char[] resultArray = new char[input.Length];
```

Comment:
```suggestion
        var resultArray = new char[input.Length];
```
не пишите тип дважды